plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -3185,12 +3199,32 @@ protected String getIoUringCheckCommand() {
      * (i) Qemu >= 5.0;
      * (ii) Libvirt >= 6.3.0
      */
-    protected void setDiskIoDriver(DiskDef disk) {
-        if (enableIoUring) {
+    public void setDiskIoDriver(DiskDef disk, boolean iothreadsEnabled, IoDriver ioDriver) {
+        if (iothreadsEnabled) {
+            disk.setIothreads(iothreadsEnabled);
+            if (!enableIoUring && IoDriver.IOURING == ioDriver) {
```

Comment:
Should we maybe remove this check for iouring? That this PR simply allows you to enable io_uring per VM/Storage pool? Remove the agent.properties setting.

Reply:
@wido, I think it's good to leave the check if the OS supports the `io_uring` option. I've just removed it from the `agent.properties` file.